lib/actions/usbsdfuTargetActions.js

```diff
@@ -415,6 +417,21 @@ function operateDFU(inputDfuImages, inputDevice) {
         // Catch error during dfu
         .catch(async error => {
             logger.error(`DFU failed: ${error.message || error}`);
+
+            // Notify the user to choose writable firmwares
+            if (error.message.includes(
+                'The array of supported SoftDevices for the update does not ' +
+                'contain the FWID of the current SoftDevice.')) {
+                dispatch(warningActions.addUserWarning(
+                    'SdReq is not supported. Please select a correct one.'));
+            }
+
+            // Notify the user to choose writable Bootloader
+            if (error.message.includes('This bootloader requires DFU updates to be signed.')) {
+                dispatch(warningActions.addUserWarning(
+                    'This bootloader need to be signed. Your action is rejected.'));
```

Comment:
Suggest rewording to something like "The bootloader update must be signed. The DFU operation has been cancelled."